pkg/prometheus/prometheus.go

```diff
@@ -43,37 +42,37 @@ func (p PrometheusClient) GetUptime(checkKey, duration string) (float64, error)
 	return 100 - float64(uptime.(model.Vector)[0].Value), nil
 }
 
-func NewPrometheusAPI(ctx dutyContext.Context, url string, auth *v1.PrometheusAuth) (*PrometheusClient, error) {
+func NewPrometheusAPI(ctx dutyContext.Context, url string, auth types.Authentication) (*PrometheusClient, error) {
 	if url == "" {
 		return nil, nil
 	}
 
 	roundTripper := prometheusapi.DefaultRoundTripper
-	if auth != nil {
-		username, err := ctx.GetEnvValueFromCache(auth.Basic.Username, ctx.GetNamespace())
+	if !auth.Username.IsEmpty() {
+		username, err := ctx.GetEnvValueFromCache(auth.Username, ctx.GetNamespace())
 		if err != nil {
 			return nil, err
 		}
 
-		password, err := ctx.GetEnvValueFromCache(auth.Basic.Password, ctx.GetNamespace())
+		password, err := ctx.GetEnvValueFromCache(auth.Password, ctx.GetNamespace())
 		if err != nil {
 			return nil, err
 		}
 
-		if auth.Basic != nil {
+		if !auth.Username.IsEmpty() {
 			roundTripper = prometheusConfig.NewBasicAuthRoundTripper(
 				username,
 				prometheusConfig.Secret(password),
 				"",
 				"",
 				roundTripper)
-		} else if auth.Oauth != nil {
-			clientID, err := ctx.GetEnvValueFromCache(auth.Oauth.ClientID, ctx.GetNamespace())
+		} else if !auth.OAuth.IsEmpty() {
```

Comment:
I think this if is too nested, it requires username to be not empty (which triggers the higher if) and this condition is never called.

Reply:
Simplified to single nesting